source/common/config/subscription_base.h

```diff
@@ -2,15 +2,21 @@
 
 #include "envoy/config/subscription.h"
 
+#include "common/config/opaque_resource_decoder_impl.h"
 #include "common/config/resource_name.h"
 
 namespace Envoy {
 namespace Config {
 
-template <typename Current> struct SubscriptionBase : public Config::SubscriptionCallbacks {
+template <typename Current>
+struct SubscriptionBase : public Config::SubscriptionCallbacks,
+                          public Config::OpaqueResourceDecoderImpl<Current> {
 public:
-  SubscriptionBase(const envoy::config::core::v3::ApiVersion api_version)
-      : api_version_(api_version) {}
+  SubscriptionBase(const envoy::config::core::v3::ApiVersion api_version,
+                   ProtobufMessage::ValidationVisitor& validation_visitor,
+                   absl::string_view name_field)
+      : OpaqueResourceDecoderImpl<Current>(validation_visitor, name_field),
```

Comment:
maybe this should be a member variable -vs- a base class?

Reply:
Good idea, updated.